### DISCUSSION TITLE
컴퓨터 네트워크(가은-> 진)

### DISCUSSION THREAD

**Discussion by gaeunnlee** [tempdku/temp#7] [General]
### 컴퓨터 네트워크 질의 응답 
@jjinleee

1. REST API가 무엇인지 설명하고, 작동 방식을 설명해주세요.

2. HTTP/2와 HTTP/3의 차이점에 대하여 설명해주세요.

**Comment by jjinleee** [comment]
### 1. REST API가 무엇인지 설명하고, 작동 방식을 설명해주세요.
REST API란 애플리케이션이나 디바이스가 연결되어 통신할 수 있는 방법을 정의한 규칙으로  REST 아키텍쳐 스타일의 디자인 원칙을 준수하는 API다. 작동 방식은 HTTP URI를 통해 자원명시, HTTP method를 통해, 해당자원에 대한 CRUD operation을 적용하며 상태코드를 포함한다.
상태를 저장하지 않고 요청간 의존성이 없도록 설계되어, 서버의 부하를 줄이고 확장성을 높일 수 있다

----

### 2. HTTP/2와 HTTP/3의 차이점에 대하여 설명해주세요.

HTTP/2와 HTTP/3는 모두 HTTP 프로토콜의 다음 버전을 나타낸다
*  프로토콜의 기반:  HTTP/2-TCP, HTTP/3-UDP
* 다중화 및 병렬 전송: HTTP/2-단일, HTTP/3-QUIC 다중화
* 헤더 압축 방식: HTTP/2-HPACK 알고리즘, HTTP/3-QUIC 헤더 압축 알고리즘
* 대기 시간 감소 : HTTP/3는 핸드셰이크 연결과정을 거치지 않음
* 서버 푸시 수정 :HTTP/2는 지원, HTTP/3는 지원x
